### ISSUE TITLE
AppCleaner leaving cache behind

### ISSUE THREAD

**Issue by Taz112768** [d4rken-org/sdmaid-se#588]
After scan I am told there is residual files that can be deleted but never deleted them. Screen shot below
![Screenshot_20230812-082350](https://github.com/d4rken-org/sdmaid-se/assets/68667867/c316be1a-b797-4c80-9548-55fb4486eb8f)

**Describe the bug**

**To Reproduce**

**Expected behavior**

**Smartphone (please complete the following information):**
 - Device: [e.g. Pixel 9]
 - OS: [e.g. Android 17]
 - Root status [e.g. Rooted - Magisk v25.2]
 - SD Maid 2/SE version [e.g. v0.6.6-rc6]

**Debug log**
If you can reproduce the bug, record the event in [a debug log](https://github.com/d4rken-org/sdmaid-se/wiki/Bugs#debug-log).

**Screenshots/Video**
If applicable, add screenshots or a video if it helps to explain your problem.

**Additional context**

**Comment by d4rken** [comment]
Do you mean AppCleaner or SystemCleaner?

AppCleaner could be stale data being reported by the system APIs.
SystemCleaner could be a permission issue.

Please provide a debug log of this.

**Comment by demetry14** [comment]
AppCleaner and I have the same issue, but I am not rooted per the main reporters description and I have tried Shizuku for elevated privilege without success also.

https://github.com/d4rken-org/sdmaid-se/assets/5199569/d4eef5ee-7243-45bd-ae4e-a0d2653a5675

**Comment by d4rken** [comment]
Do you think that SD Maid is not pressing the clear cache button? Or that pressing the button has no effect?

**Comment by demetry14** [comment]
> Do you think that SD Maid is not pressing the clear cache button? Or that pressing the button has no effect?

I tested the system Clear Cache button and it looks like the system is not clearing cache when the button is pressed for that app.

**Comment by d4rken** [comment]
Then SD Maid can't really do it differently... It would be interesting if it makes a difference whether "force stop" is pressed before "clear cache" as this is something SD Maid could do too.

**Comment by MarkyMarkDE** [comment]
have exact the same problem since SD Maid 1. It runs (so i think) every evening at 8pm and when I after open the App I get the Info "xx MB / GB" - but why when the "cron job" has previous runned successfully? OK, then i open SD Maid again and see much often the same value which can be deleted ... 
see Ticket [#562](https://github.com/d4rken-org/sdmaid-se/issues/562)

**Comment by lawson85** [comment]
https://stackoverflow.com/questions/25631612/how-to-use-the-adb-shell-pm-trim-caches-command

**Comment by d4rken** [comment]
> have exact the same problem since SD Maid 1. It runs (so i think) every evening at 8pm and when I after open the App I get the Info "xx MB / GB" can be deleted - but why when the "cron job" has previous runned successfully? OK, then i open SD Maid again and see much often the same value which can be deleted ... I'm not sure if the cron run successful or what exact are the reason for this Problem
see Ticket https://github.com/d4rken-org/sdmaid-se/issues/562

Mhhh not sure if this is something else:

Background execution via scheduler does not execute the accessibility service based deletion, as this can not happen in the background or with the screen off/locked.

> https://stackoverflow.com/questions/25631612/how-to-use-the-adb-shell-pm-trim-caches-command

If Shizuku is set up, then SD Maid already uses this. Only works with user apps though, and only in "delete all" mode, i.e. you can't target individual apps with that.

The issue here by @Taz112768 and @demetry14 is a different one as far as I could grasp it: Pressing the clear cache button has no (or little effect). More of a system issue than SD Maid issue, but I'm still interested in potential workarounds.

**Comment by MarkyMarkDE** [comment]
@d4rken okay maybe it is not identically, but the main problem still occurs, how often I delete the caches SD Maid 1 + 2 delete nothing (or it looks like) and I mean not when SD Maid is in background ... open SD Maid 2 "XX MB/GB" can be deleted then I start run to delete the Caches, after that I close SD Maid 2 and open it again and scan for cache size and mostly the same amount of delete size will be shown as before the previous deleting ...

**Comment by MarkyMarkDE** [comment]
> > have exact the same problem since SD Maid 1. It runs (so i think) every evening at 8pm and when I after open the App I get the Info "xx MB / GB" can be deleted - but why when the "cron job" has previous runned successfully? OK, then i open SD Maid again and see much often the same value which can be deleted ... I'm not sure if the cron run successful or what exact are the reason for this Problem
> > see Ticket #562
> 
> Mhhh not sure if this is something else:
> 
> Background execution via scheduler does not execute the accessibility service based deletion, as this can not happen in the background or with the screen off/locked.
> 
> > https://stackoverflow.com/questions/25631612/how-to-use-the-adb-shell-pm-trim-caches-command
> 
> If Shizuku is set up, then SD Maid already uses this. Only works with user apps though, and only in "delete all" mode, i.e. you can't target individual apps with that.
> 
> The issue here by @Taz112768 and @demetry14 is a different one as far as I could grasp it: Pressing the clear cache button has no (or little effect). More of a system issue than SD Maid issue, but I'm still interested in potential workarounds.

system issue since really long time for ancient android versions? This issue I have since SD Maid 1 - not sure if CCleaner does work correctly for deleting the Caches ...

**Comment by bjpafa** [comment]
Agreed.

**Comment by Taz112768** [comment]
I'm closing as I'm very learned since I posted that with a unrooted phone android limits the access and has small issue accessing these areas. It's not a app issue but a phone on OS issue. It's really not a big deal as 98% of garbage delete, it is a great app I've used for years. The first one I had that does what it says and was first I had found that auto clears cache, most before Darken made this app were manual. I can verify if you root your app you will get all done ice done it but found the process made no difference in my speed with the tiny amount left and without them. Thanks for reading.
